paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/procedure/CreateTagProcedure.java

```diff
@@ -61,25 +67,45 @@ public String[] call(
         return innerCall(tableId, tagName, snapshotId, timeRetained);
     }
 
-    public String[] call(
+    public @DataTypeHint("ROW<snapshot STRING>") Row[] call(
             ProcedureContext procedureContext, String tableId, String tagName, String timeRetained)
             throws Catalog.TableNotExistException {
         return innerCall(tableId, tagName, null, timeRetained);
     }
 
-    private String[] innerCall(
+    private @DataTypeHint("ROW<snapshot STRING>") Row[] innerCall(
             String tableId,
             String tagName,
             @Nullable Long snapshotId,
             @Nullable String timeRetained)
             throws Catalog.TableNotExistException {
-        Table table = catalog.getTable(Identifier.fromString(tableId));
+        FileStoreTable fileStoreTable =
+                (FileStoreTable) catalog.getTable(Identifier.fromString(tableId));
+        SnapshotManager snapshotManager = fileStoreTable.snapshotManager();
+        TagManager tagManager = fileStoreTable.tagManager();
+        Snapshot fromSnapshot = null;
         if (snapshotId == null) {
-            table.createTag(tagName, toDuration(timeRetained));
+            fromSnapshot = snapshotManager.latestSnapshot();
         } else {
-            table.createTag(tagName, snapshotId, toDuration(timeRetained));
+            try {
+                fromSnapshot = tagManager.createTagInternal(snapshotId, snapshotManager);
+            } catch (Exception ignored) {
```

Comment:
Here should not catch all exceptions.

Maybe we should refactor `createTagInternal` return value or throw a special exception.

Reply:
Just catch `IllegalArgumentException`.

Reply:
You could create a new Exception class. For example, `SnapshotNotExistException`